simplified-cardcreator/src/main/java/org/nypl/simplified/cardcreator/ui/LocationFragment.kt

```diff
@@ -0,0 +1,263 @@
+   *
+   * @param requestCode - String user defined request code to identify the request
+   * @param permissions - String Array of permissions requested
+   * @param grantResults - Integer array of what the user has granted/denied
+   */
+  override fun onRequestPermissionsResult(
+    requestCode: Int,
+    permissions: Array<String>,
+    grantResults: IntArray
+  ) {
+    when (requestCode) {
+      locationRequestCode -> {
+        // If request is cancelled, the result arrays are empty.
+        if ((grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED)) {
+          logger.debug("Location permission granted")
+
+        } else {
+          logger.debug("Location permission NOT granted")
+        }
+        return
+      }
+    }
+  }
+
+  override fun onLocationChanged(location: Location?) {
+    logger.debug("Location has changed")
+    val locationManager = activity!!.getSystemService(Context.LOCATION_SERVICE) as LocationManager
+    if (location != null) {
+      logger.debug("Checking to see if user is in New York")
+      showLoading(true)
+      binding.nextBtn.isEnabled = false
+      val maxResults = 1
+      val geocoder = Geocoder(activity!!, Locale.getDefault())
+
+      // Address found using the Geocoder.
+      val address: Address?
+
+      try {
+          address = geocoder.getFromLocation(location.latitude, location.longitude, maxResults)[0]
+          logger.debug( "Region is: ${address.adminArea} ${address.countryCode} ")
+          binding.regionEt.setText("${address.adminArea} ${address.countryCode}", TextView.BufferType.EDITABLE)
+
+          if (address.countryCode == "US" && (address.adminArea == "New York" || address.adminArea == "NY")) {
+            logger.debug("User is in New York")
+            isNewYork = true
+          }
+      } catch (e: Exception) {
+        logger.debug("Error checking to see if user is in New York")
+        e.printStackTrace()
+      }
+      showLoading(false)
+      // Show error if user is not in New York
+      if (!isNewYork) {
+        logger.debug("User is NOT in New York")
+        binding.nextBtn.isEnabled = true
+        binding.nextBtn.text = getString(R.string.done)
+        binding.headerStatusDescTv.text = getString(R.string.location_error)
+        binding.headerStatusDescTv.setTextColor(ContextCompat.getColor(activity!!, R.color.red))
+      } else {
+        binding.nextBtn.isEnabled = true
+        binding.nextBtn.text = getString(R.string.next)
+        binding.headerStatusDescTv.text = getString(R.string.new_york_success)
+        binding.headerStatusDescTv.setTextColor(ContextCompat.getColor(activity!!, R.color.trans_black))
+      }
+    }
+    logger.debug("Stopping location updates")
+    locationManager.removeUpdates(this)
+  }
+
+  // These are not needed but were invited to the party by Google
+  override fun onStatusChanged(provider: String?, status: Int, extras: Bundle?) { TODO("Not yet implemented") }
```

Comment:
This will cause the app to crash with a `NotImplementedError` if a location provider's status changes during our request. Unlikely, but since we can't control when the system invokes these callbacks we should remove these calls to `TODO(...)`.